driver does no  longer compile with kernel 4.15

To solve the problem that my realtek RTL8812AE 802.11ac chip does not support the 5GHz band i used  to compile the drivers provided here.

However, with kernel 4.15 (unstable branch) this does no longer work. I get the following error

    make[1]: Verzeichnis „/usr/lib/modules/4.15.0-1-MANJARO/build“ wird betreten
    CC [M] /home/xxx/treiber/rtlwifi_new/base.o
    In file included from ./arch/x86/include/asm/percpu.h:45:0,
    from ./arch/x86/include/asm/current.h:6,
    from ./include/linux/sched.h:12,
    from /home/xxx/treiber/rtlwifi_new/wifi.h:31,
    from /home/xxx/treiber/rtlwifi_new/base.c:26:
    ./include/linux/kernel.h:6:10: schwerwiegender Fehler: stdarg.h: Datei oder Verzeichnis nicht gefunden
    #include <stdarg.h>
    ^~~~~~~~~~

that is, the file <stdarg.h> is not found. 

I do not know if this is specific for the arch-based Manjaro distribution or a general problem with kernel 4.15.

Similar problems running OpenSUSE Tumbleweed after upgrade to kernel 4.15:
```
> make clean && make KVER=4.15.0-1-default
make -C /lib/modules/4.15.0-1-default/build M=/home/adam/Downloads/rtlwifi_new modules
make[1]: Verzeichnis „/usr/src/linux-4.15.0-1-obj/x86_64/default“ wird betreten
  CC [M]  /home/adam/Downloads/rtlwifi_new/base.o
/home/adam/Downloads/rtlwifi_new/base.c: In function ‘_rtl_init_deferred_work’:
/home/adam/Downloads/rtlwifi_new/base.c:576:2: error: implicit declaration of function ‘setup_timer’; did you mean ‘sk_stop_timer’? [-Werror=implicit-function-declaration]
  setup_timer(&rtlpriv->works.watchdog_timer,
  ^~~~~~~~~~~
  sk_stop_timer
cc1: some warnings being treated as errors
```

Could this be related to the [recent change](https://old.lwn.net/Articles/735887/) in the kernel's timer API?

same problem on extended branch, just upgraded to 4.15.15-1-ARCH

```
@ember:rtlwifi_new$ git status
On branch extended
Your branch is up to date with 'origin/extended'.

nothing to commit, working tree clean
b@ember:rtlwifi_new$ make clean
b@ember:rtlwifi_new$ make
make -C /lib/modules/4.15.15-1-ARCH/build M=/data/GitHub/rtlwifi_new modules
make[1]: Entering directory '/usr/lib/modules/4.15.15-1-ARCH/build'
  CC [M]  /data/GitHub/rtlwifi_new/base.o
In file included from ./arch/x86/include/asm/percpu.h:45:0,
                 from ./arch/x86/include/asm/current.h:6,
                 from ./include/linux/sched.h:12,
                 from /data/GitHub/rtlwifi_new/wifi.h:31,
                 from /data/GitHub/rtlwifi_new/base.c:26:
./include/linux/kernel.h:6:10: fatal error: stdarg.h: No such file or directory
 #include <stdarg.h>
          ^~~~~~~~~~
compilation terminated.
```

--- a/Makefile
+++ b/Makefile
@@ -160,7 +160,7 @@ endif
 clean:
 	@find halmac/ \( -name "*.mod.c" -o -name "*.mod" -o -name "*.o" -o -name ".*.cmd" -o -name "*.ko" -o -name "*~" \) -exec rm {} \;
 	@find phydm/ \( -name "*.mod.c" -o -name "*.mod" -o -name "*.o" -o -name ".*.cmd" -o -name "*.ko" -o -name "*~" \) -exec rm {} \;
-	@rm -fr *.mod.c *.mod *.o .*.cmd *.ko *~
+	@rm -fr *.mod.c *.mod *.o .*.cmd *.ko *~ .cache.mk
 	@rm -fr rtl8*/*.mod.c rtl8*/*.mod rtl8*/*.o rtl8*/.*.cmd rtl8*/*.ko rtl8*/*~
 	@rm -fr bt*/*.mod.c bt*/*.mod bt*/*.o bt*/.*.cmd bt*/*.ko bt*/*~
 	@rm -fr .tmp_versions

